### PR TITLE
Migrate to Vite + Refactor code [Needs review]

### PR DESCRIPTION
**Pull request by Posandu**
# Description

Improved the code and migrated to Vite.

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected) 
- [x] This change requires a documentation update

### CODE REVIEW

**Comment by yashash1511** [comment]
very nic this was very much required else the development for win11React is very slow.

also don't worry about the deploy preview builds failing i will fix them just let me know when it is ready

**Comment by Posandu** [comment]
@yashash1511 I'm done but I can't do the serviceworker stuff.
image

**Comment by yashash1511** [comment]
is it complete?

**Comment by Posandu** [comment]
nvm fixed it

**Comment by yashash1511** [comment]
> * [x]  Breaking change (fix or feature that would cause existing functionality to not work as expected)

what is it breaking @Posandu ?

**Comment by Posandu** [comment]
The translations system should be fixed

**Comment by yashash1511** [comment]
> The translations system should be fixed

ohhk

**Comment by Posandu** [comment]
Oh then it works

**Comment by yashash1511** [comment]
very nic build time also down to 16s from 38s earlier

File: index.html [hunk 1]
```diff
@@ -4,7 +4,7 @@
     <meta charset="UTF-8" />
     <meta http-equiv="X-UA-Compatible" content="IE=edge" />
     <meta name="viewport" content="width=device-width, initial-scale=1.0" />
-    <link rel="icon" href="favicon.ico" />
+    <link rel="icon" href="/favicon.ico" />
```
Comment (yashash1511) [inline index.html:7]: don't add these '/'

Comment (yashash1511) [reply index.html:7]: nvm

**Review by yashash1511 — APPROVED** [review]
LGTM

**Pull request merged**